src/transformers/models/llama/modeling_llama.py

```diff
@@ -869,22 +948,8 @@ def forward(
 
         if inputs_embeds is None:
             inputs_embeds = self.embed_tokens(input_ids)
-        # embed positions
```

Comment:
We don't need to unnecessarily create attention mask if not needed. E.g. the 4D attention_mask is never needed for FA2